lib/ddtrace/utils/forking.rb

```diff
@@ -43,9 +43,16 @@ module ClassExtensions
         # object will cause forking to not be detected in the fork when it should have.
         #
         # This wrapper prevents this by initializing the fork PID when the object is created.
-        def initialize(*args, &block)
-          super
-          update_fork_pid!
+        if RUBY_VERSION >= '3'
```

Comment:
You have the option to use this syntax as well to safely support delegation in all versions of Ruby:
https://github.com/DataDog/dd-trace-rb/blob/e391d2eb64d3c6151a4bdd2710c6a8c7c1d57457/lib/ddtrace/contrib/sucker_punch/instrumentation.rb#L42-L52
Up to you.

Reply:
I think I find the existing version cleaner looking. It's never really clear or intuitive to me what `ruby2_keywords` does, despite it being recommended by the Ruby docs.